### PR TITLE
DEV-1035: Add support for skipping nodes during tagging/versioning

### PR DESCRIPTION
**Pull request by kulgan**
Add implementation that ignores nodes with certain values for certain fields while generating a tag. These fields are configured via the dictionary. A sample configuration is also included for testing purposes.
* Adds is_taggable method to every node that supports versioning. This returns true only if a tag can be generated for the node
* Omit tag generation for nodes with is_taggable false
* Also omit parent nodes with is_taggable false while generating tag for child nodes

### CODE REVIEW

File: gdcdatamodel/models/__init__.py [hunk 1]
```diff
@@ -373,6 +374,18 @@ def tag_properties(self):
         """
         return tag_props
 
+    @property
+    def tagging_config(self):
+        """Tagging configuration instance used for checking if node instance participates in versioning
+
+        Returns:
+            versioning.TaggingConfig): config instance
```
Comment (mpsolano) [inline gdcdatamodel/models/__init__.py:382]:
```suggestion
            versioning.TaggingConfig: config instance
```

File: gdcdatamodel/models/versioning.py [hunk 2]
```diff
@@ -22,6 +21,86 @@ class TagKeys:
...
+            node (psqlgraph.Node): node instance
+        Returns:
+            Returns (bool)
+        """
+        node = self._resolve_target_node_from_path(node)
+        return node[self.prop] in self.values
+
+
+class TaggingConfig:
+    """A wrapper around the tagConfig definition in the dictionary yaml"""
```
Comment (mpsolano) [inline gdcdatamodel/models/versioning.py:76]: Should the docstring reference tagBuilderConfig?

**Review by mpsolano — APPROVED** [review]
Looks good!

File: gdcdatamodel/models/__init__.py [hunk 3]
```diff
@@ -373,6 +374,18 @@ def tag_properties(self):
...
+        """
+        return versioning.TaggingConfig(cfg=tag_config)
+
+    def is_taggable(self):
+        return self.tagging_config.is_taggable(self)
```
Comment (qiaouchicago) [inline gdcdatamodel/models/__init__.py:387]: `self`?

File: gdcdatamodel/models/versioning.py [hunk 4]
```diff
@@ -85,6 +164,10 @@ def inject_set_tag_after_insert(cls):
     @event.listens_for(cls, "after_insert")
     def set_node_tag(mapper, conn, node):
         table = node.__table__
+
+        if not node.is_taggable():
+            return  # do nothing
+
```
Comment (XukunXiangUChicago) [inline gdcdatamodel/models/versioning.py:170]: nit picking:
```suggestion
        if not node.is_taggable():
            return  # do nothing
            
        table = node.__table__
```

**Review by XukunXiangUChicago — APPROVED** [review]

File: gdcdatamodel/models/__init__.py [hunk 5]
```diff
@@ -17,6 +17,8 @@
 import os
 import sys
 
+import six
```
Comment (qiaouchicago) [inline gdcdatamodel/models/__init__.py:20]: We are still supporting python 2?

Comment (kulgan) [reply gdcdatamodel/models/__init__.py:20]: Sadly we are, [biowcs](https://github.com/NCI-GDC/biowcs) is still py2 and it is a mojor player for GPAS

**Pull request merged**